feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go

```diff
@@ -0,0 +1,779 @@
+	desc        string
+	applyPolicy func(t *testing.T, dut *ondatra.DUTDevice)
+	validate    func(t *testing.T, dut *ondatra.DUTDevice, ate *ondatra.ATEDevice)
+	ipv4        bool
+	flowConfig  flowConfig
+}
+
+type flowConfig struct {
+	src   attrs.Attributes
+	dstNw string
+	dstIP string
+}
+
+func TestBGPNestedPolicies(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	configureDUT(t, dut)
+
+	ate := ondatra.ATE(t, "ate")
+	top := gosnappi.NewConfig()
+	devs := configureOTG(t, ate, top)
+	td := testData{
+		dut:   dut,
+		ate:   ate,
+		top:   top,
+		otgP1: devs[0],
+		otgP2: devs[1],
+	}
+	td.advertiseRoutesWithEBGP(t)
+	ate.OTG().PushConfig(t, top)
+	ate.OTG().StartProtocols(t)
+	defer ate.OTG().StopProtocols(t)
+	otgutils.WaitForARP(t, ate.OTG(), top, "IPv4")
+	otgutils.WaitForARP(t, ate.OTG(), top, "IPv6")
+	td.verifyDUTBGPEstablished(t)
+	td.verifyOTGBGPEstablished(t)
+	td.verifyNestedImportPolicyAttachedv4(t)
+	td.verifyNestedImportPolicyAttachedv6(t)
+	td.verifyNestedExportPolicyAttachedv4(t)
+	td.verifyNestedExportPolicyAttachedv6(t)
+
+	testCases := []testCase{
+		{
+			name:        "IPv4BGPNestedmportPolicy",
+			desc:        "IPv4 BGP Nested import policy test",
+			applyPolicy: configureImportRoutingPolicy,
+			validate:    validateImportRoutingPolicy,
+			ipv4:        true,
+			flowConfig:  flowConfig{src: atePort2, dstNw: "v4-bgpNet-dev1", dstIP: v41TrafficStart},
+		},
+		{
+			name:        "IPv4BGPNestedExportPolicy",
+			desc:        "IPv4 BGP Nested export policy test",
+			applyPolicy: configureExportRoutingPolicy,
+			validate:    validateExportRoutingPolicy,
+			ipv4:        true,
+			flowConfig:  flowConfig{src: atePort1, dstNw: "v4-bgpNet-dev2", dstIP: v42TrafficStart},
+		},
+		{
+			name:        "IPv6BGPNestedImportPolicy",
+			desc:        "IPv6 BGP Nested import policy test",
+			applyPolicy: configureImportRoutingPolicyV6,
+			validate:    validateImportRoutingPolicyV6,
+			ipv4:        false,
+			flowConfig:  flowConfig{src: atePort2, dstNw: "v6-bgpNet-dev1", dstIP: v61TrafficStart},
+		},
+		{
+			name:        "IPv6BGPNestedExportPolicy",
+			desc:        "IPv6 BGP Nested export policy test",
+			applyPolicy: configureExportRoutingPolicyV6,
+			validate:    validateExportRoutingPolicyV6,
+			ipv4:        false,
+			flowConfig:  flowConfig{src: atePort1, dstNw: "v6-bgpNet-dev2", dstIP: v62TrafficStart},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Logf("Description: %s", tc.desc)
+			tc.applyPolicy(t, dut)
+			tc.validate(t, dut, ate)
+			if tc.ipv4 {
+				createFlow(t, td, tc.flowConfig)
+				checkTraffic(t, td, v4Flow)
+			} else {
+				createFlowV6(t, td, tc.flowConfig)
+				checkTraffic(t, td, v6Flow)
+			}
+		})
+	}
+}
+
+// IPv4 BGP nested import policy test
+func configureImportRoutingPolicy(t *testing.T, dut *ondatra.DUTDevice) {
+	root := &oc.Root{}
+	rp := root.GetOrCreateRoutingPolicy()
+
+	//Configure a route-policy to set the local preference
```

Comment:
here and elsewhere:
https://google.github.io/styleguide/go/decisions#comment-sentences

If the comment are complete sentence, they should capitablized and punctuated 
also you need to space after //
for example // Configure a route-policy to set the local preference.